### PR TITLE
Something  wrong with getting relevant 

### PR DESCRIPTION
**Pull request by zhaosijun**
The "ActiveDeployment" return the address of copy element from "input".  In my opinion, It  should be the address of element from "input" according to the line 429

### CODE REVIEW

File: pkg/deploy/util/util.go [hunk 1]
```diff
@@ -41,19 +41,10 @@ func LatestDeploymentInfo(config *deployapi.DeploymentConfig, deployments []api.
...
 // latest deployment.
 func ActiveDeployment(config *deployapi.DeploymentConfig, input []api.ReplicationController) *api.ReplicationController {
-	// we need to create our own copy of the input slice so that our sort here
-	// doesn't change the caller's slice
-	var deployments []api.ReplicationController
-	for _, deployment := range input {
-		deployments = append(deployments, deployment)
-	}
-
-	sort.Sort(ByLatestVersionDesc(deployments))
```
Comment (kargakis) [inline pkg/deploy/util/util.go]: There are no guarantees that `input` is sorted and we always need to return the latest complete deployment.

File: pkg/deploy/util/util.go [hunk 2]
```diff
@@ -425,9 +416,9 @@ func DeploymentsForCleanup(configuration *deployapi.DeploymentConfig, deployment
 	} else {
 		// if there is an active deployment, we need to filter out any deployments that we don't
 		// care about, namely the active deployment and any newer deployments
-		for _, deployment := range deployments {
-			if &deployment != activeDeployment || DeploymentVersionFor(&deployment) < DeploymentVersionFor(activeDeployment) {
-				relevantDeployments = append(relevantDeployments, deployment)
+		for i := range deployments {
+			if &deployments[i] != activeDeployment && DeploymentVersionFor(&deployments[i]) < DeploymentVersionFor(activeDeployment) {
```
Comment (kargakis) [inline pkg/deploy/util/util.go]: Can you also make the same change for the loop above? Even more readable:
```go
    for i := range deployments {
          deployment := &deployments[i]
 			if DeploymentVersionFor(deployment) != configuration.Status.LatestVersion {
 				relevantDeployments = append(relevantDeployments, *deployment)
 			}
    }
} else {
 		// if there is an active deployment, we need to filter out any deployments that we don't
 		// care about, namely the active deployment and any newer deployments
       for i := range deployments {
          deployment := &deployments[i]
		  if deployment != activeDeployment && DeploymentVersionFor(deployment) < DeploymentVersionFor(activeDeployment) {
                 relevantDeployments = append(relevantDeployments, deployment)
```

Comment (zhaosijun) [inline pkg/deploy/util/util.go]: @kargakis In “ActiveDeployment” compare the "deploymentVersion" should return the latest complete deployment and not change the "input". Thanks.

**Comment by zhaosijun** [comment]
@kargakis PTAL

**Comment by kargakis** [comment]
LGTM

**Comment by kargakis** [comment]
[test]

**Comment by kargakis** [comment]
Please squash down to one commit and simplify the commit message to something like "Switch deployment utils to loop by index"

**Pull request closed without merging**